### PR TITLE
Added extra fees dialogue

### PR DESCRIPTION
**Pull request by zaelgohary**
### Description

- Added a new action in your farm nodes section
- Added extra fees dialogue

image

image

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/499

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

File: packages/dashboard/src/portal/components/FarmNodesTable.vue [hunk 1]
```diff
@@ -314,6 +323,45 @@
...
+          <v-card-text class="text">
+            <v-form v-model="isValidExtraFee">
+              <v-text-field
+                class="mt-4"
+                label="Extra Fees"
+                v-model="extraFee"
+                required
+                outlined
+                dense
+                hint="Extra fees are in USD"
```
[the remark this answers is not in the document]
Comment (xmonader) [reply packages/dashboard/src/portal/components/FarmNodesTable.vue:340]: In general the GPU is already very expensive to be expressed in milli USD, it's already expressed in general with USD/hr on other cloud providers. Let's put it USD/Month instead for the farmer

**Comment by xmonader** [comment]
I suggest 
- we use `Set Fees` or `Set Additional Fees`  as a title (tooltip/explanation) most useful to set the price for GPU or special hardware added to the node (or something like that)
the unit should be on the right so I guest

- field row to be `[  10    ] USD/month (tooltip)` about the payment going to happen in TFT nevertheless but with price anchored to USD, or in the field label itself we say it's USD/month

File: packages/dashboard/src/portal/components/FarmNodesTable.vue [hunk 2]
```diff
@@ -629,6 +686,27 @@ export default class FarmNodesTable extends Vue {
     }
     this.openPublicConfigDialog = true;
   }
+
+  async openExtraFee(node: nodeInterface) {
+    this.nodeToEdit = node;
+    // convert fees from USD to mili USD while getting
+    this.extraFee =
+      (await this.queryClient.contracts.getDedicatedNodeExtraFee({ nodeId: this.nodeToEdit.nodeId })) / 1000;
```
Comment (AhmedHanafy725) [inline packages/dashboard/src/portal/components/FarmNodesTable.vue:694]: the value should be multiplied by 1000 not divided

File: packages/dashboard/src/portal/components/FarmNodesTable.vue [hunk 3]
```diff
@@ -629,6 +686,27 @@ export default class FarmNodesTable extends Vue {
...
+  }
+
+  async saveExtraFee(fee: number) {
+    this.loadingExtraFee = true;
+    // convert fees from mili USD to USD while setting
+    setDedicatedNodeExtraFee(this.$store.state.credentials.account.address, this.nodeToEdit.nodeId, fee * 1000).then(
+      () => {
+        this.$toasted.show(`Transaction succeeded: Fee is added to node ${this.nodeToEdit.nodeId}`);
+        this.loadingExtraFee = false;
+        this.openExtraFeeDialogue = false;
```
Comment (AhmedHanafy725) [inline packages/dashboard/src/portal/components/FarmNodesTable.vue:705]: can we also handle the failure case?
it keeps loading forever
image

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

File: packages/dashboard/src/portal/components/FarmNodesTable.vue (hunk 2 above)
Comment (AhmedHanafy725) [reply packages/dashboard/src/portal/components/FarmNodesTable.vue:694]: thought it was on setting not the getting, so this should be ignored

Comment (zaelgohary) [reply packages/dashboard/src/portal/components/FarmNodesTable.vue:700]: It's divided while getting & multiplied while setting.

File: packages/dashboard/src/portal/components/FarmNodesTable.vue (hunk 3 above)
Comment (zaelgohary) [reply packages/dashboard/src/portal/components/FarmNodesTable.vue]: handled.

**Pull request merged**